example/src/screens/PaymentSheetDeferredIntentMultiStepScreenWithUpdates.tsx

```diff
@@ -0,0 +1,311 @@
+      },
+    });
+    const { customer, ephemeralKey } = await response.json();
+
+    return {
+      customer,
+      ephemeralKey,
+    };
+  };
+
+  const initialisePaymentSheet = async () => {
+    setLoading(true);
+
+    try {
+      const { customer, ephemeralKey } = await fetchPaymentSheetParams();
+
+      const { error, paymentOption } = await initPaymentSheet({
+        customFlow: true,
+        customerId: customer,
+        customerEphemeralKeySecret: ephemeralKey,
+        merchantDisplayName: 'Example Inc.',
+        applePay: {
+          merchantCountryCode: 'US',
+        },
+        googlePay: {
+          merchantCountryCode: 'US',
+          testEnv: __DEV__,
+        },
+        returnURL: 'stripe-example://stripe-redirect',
+        billingDetailsCollectionConfiguration: {
+          attachDefaultsToPaymentMethod: true,
+          name: PaymentSheet.CollectionMode.ALWAYS,
+          address: PaymentSheet.AddressCollectionMode.NEVER,
+        },
+        intentConfiguration: {
+          confirmHandler: async (
+            paymentMethod: PaymentMethod.Result,
+            _shouldSavePaymentMethod: boolean,
+            intentCreationCallback: (
+              result: PaymentSheet.IntentCreationCallbackParams
+            ) => void
+          ) => {
+            const response = await fetch(
+              `${API_URL}/payment-intent-for-payment-sheet`,
+              {
+                method: 'POST',
+                headers: {
+                  'Content-Type': 'application/json',
+                },
+                body: JSON.stringify({
+                  amount: amount,
+                  currency: 'SEK',
+                  paymentMethodId: paymentMethod.id,
+                  customerId: customer,
+                }),
+              }
+            );
+            const { clientSecret, error: responseError } =
+              await response.json();
+
+            if (responseError) {
+              intentCreationCallback({
+                error: {
+                  code: 'Failed',
+                  message: responseError.raw.message,
+                  localizedMessage: responseError.raw.message,
+                },
+              });
+            } else {
+              intentCreationCallback({ clientSecret });
+            }
+          },
+          mode: {
+            amount: amount,
+            currencyCode: 'SEK',
+          },
+          paymentMethodTypes: ['card'],
+        },
+      });
+
+      if (!error) {
+        setPaymentSheetEnabled(true);
+      } else {
+        Alert.alert(`Error code: ${error.code}`, error.message);
+      }
+      if (paymentOption) {
+        setPaymentMethodOption(paymentOption);
+      }
+    } catch (error) {
+      console.log('error', error);
+    } finally {
+      setLoading(false);
+    }
+  };
+
+  const updatePayment = async () => {
+    setLoading(true);
+
+    try {
+      const { customer } = await fetchPaymentSheetParams();
```

Comment:
this might give us a new customer- should we save the old customer from the init step locally instead?